NEWS.md

```diff
@@ -99,6 +99,33 @@ This section lists changes that do not have deprecation warnings.
     `n` and `max_delay`.  The previous functionality can be achieved setting
     `delays` to `ExponentialBackOff`. ([#19331])
 
+  * `FloatRange` has been replaced by `StepRangeLen`, and the internal
+    representation of `LinSpace` has changed. Aside from changes in
+    the internal field names, this leads to several differences in
+    behavior ([#18777]):
+
+    + Both `StepRangeLen` and `LinSpace` can represent ranges of
+      arbitrary object types---they are no longer limited to
+      floating-point numbers.
+
+    + For ranges that produce `Float64`, `Float32`, or `Float16`
+      numbers, `StepRangeLen` can be used to produce values with
+      little or no roundoff error due to internal arithmetic that is
+      typically twice the precision of the output result.
+
+    + To take advantage of this precision, `linspace(start, stop,
+      len)` now returns a range of type `StepRangeLen` rather than
+      `LinSpace` when `start` and `stop` are
+      `FloatNN`. `LinSpace(start, stop, len)` always returns a
+      `LinSpace`.
+
+    + `StepRangeLen(a, step, len)` constructs a single-precision range
```

Comment:
Good point, and thanks for catching it. (I failed to locate what your first comment was in reference to, so thanks for pinging it again.) I'll wait for CI to pass and then submit that as a separate commit (not worth going through CI again).